third_party/move/move-vm/runtime/src/session.rs

```diff
@@ -60,6 +62,7 @@ impl<'r, 'l> Session<'r, 'l> {
         args: Vec<impl Borrow<[u8]>>,
         gas_meter: &mut impl GasMeter,
         traversal_context: &mut TraversalContext,
+        module_storage: &impl ModuleStorage,
```

Comment:
Would it make more sense to move the module_storage into Session? That way we could avoid changing the api. Also from the api standpoint I think it also makes sense, as we shouldn't expect modules loaded within a session to be altered.

Reply:
Not really, and this is something I do not want to do on purpose - mostly because of init-module and respawned sessions. If we capture it in Move session, then how do you support init_module? It is not possible unless you respawn session or stash published modules temporarily inside the session (which exactly lead to existing problems).

The APIs can made nice by introducing a "context" struct which you pass everywhere, but it contains the gas, the module storage, etc, something similar to `Resolver`